ota-plus-web/app/com/advancedtelematic/controllers/NamespaceController.scala

```diff
@@ -0,0 +1,53 @@
+package com.advancedtelematic.controllers
+
+import com.advancedtelematic.api.{ApiClientExec, ApiClientSupport}
+import com.advancedtelematic.auth.oidc.OidcGateway
+import com.advancedtelematic.auth.{AccessTokenBuilder, IdentityAction, SessionCodecs}
+import com.advancedtelematic.libats.data.DataType.Namespace
+import javax.inject.{Inject, Singleton}
+import play.api.libs.json._
+import play.api.libs.ws.WSClient
+import play.api.mvc._
+import play.api.{Configuration, Logger}
+
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class NamespaceController @Inject()(val conf: Configuration,
+                                    val ws: WSClient,
+                                    val clientExec: ApiClientExec,
+                                    authAction: IdentityAction,
+                                    accessTokenBuilder: AccessTokenBuilder,
+                                    oidcGateway: OidcGateway,
+                                    components: ControllerComponents)(implicit exec: ExecutionContext)
+  extends AbstractController(components) with ApiClientSupport {
+
+  private val log = Logger(this.getClass)
+  private val redirectToIndex = Redirect(com.advancedtelematic.controllers.routes.Application.index())
+
+  implicit def namespaceBinder: PathBindable[Namespace] =
+    new PathBindable[Namespace] {
+      override def bind(key: String, ns: String): Either[String, Namespace] = Right(Namespace(key))
+      override def unbind(key: String, ns: Namespace): String = ns.get
+  }
+
+  def switchNamespace(namespace: Namespace): Action[AnyContent] = authAction.async { request =>
+    oidcGateway.getUserInfo(request.accessToken).flatMap { claims =>
```

Comment:
I would suggest to take `userId` from an `IdToken` in request

Reply:
Updated.